sip_bf_detector/sip_bf_detector.cpp

```diff
@@ -89,6 +89,16 @@ trap_module_info_t *module_info = NULL;
    PARAM('o', "ok_count", "Number of observed OK responses after crossing alert threshold to consider the alert false and drop the communication. (5 by default)", required_argument, "uint32") \
    PARAM('s', "scan_threshold", "Number of extensions a specific client attempted to register as (on one particular server) which is considered a scan. (5 by default)", required_argument, "uint32")
 
+#define max(a,b) \
```

Comment:
I have some thoughts about this patch:
I wondered why you needed to make a copy of arguments into some local variables and why it's so complicated.

I've investigated a bit and I can'ŧ see any reason why we have `ur_time_t time_stamp;` at https://github.com/CESNET/Nemea-Detectors/blob/b618f56d5ffa120d83c1cdbf7584efdf4e46538a/sip_bf_detector/sip_bf_detector.h#L127
when we store `ur_time_get_sec()` into it at https://github.com/CESNET/Nemea-Detectors/blob/b618f56d5ffa120d83c1cdbf7584efdf4e46538a/sip_bf_detector/sip_bf_detector.cpp#L1590

`ur_time_get_sec()` returns `uint32_t` so this patch always compares `uint32_t` and `ur_time_t` which is not necessary (in fact, we need to compare just two 32b number of seconds)...

I'd suggest to rewrite the patch to something like:

```c
#define SETMIN(A, B) (A) = ((A) <= (B)) ? (A) : (B))
#define SETMAX(A, B) (A) = ((A) >= (B)) ? (A) : (B))
```

A minor objection is about coding style: https://nemea.liberouter.org/developer/coding-style-c#naming -> macros and their parameters should be uppercase.

Generally, it is strange to me that we use just `TIME_FIRST` and we completely avoid `TIME_LAST` but this is probably out of scope of this patch, isn'ŧ it?

Reply:
The reason the module always uses TIME_FIRST is a consequence of the input data. SIP flow always represent just one request/response between devices. Hence the statement TIME_FIRST = TIME_LAST holds always true.